genesyscloud/tfexporter/genesyscloud_resource_exporter.go

```diff
@@ -479,6 +483,49 @@ func (g *GenesysCloudResourceExporter) generateOutputFiles() diag.Diagnostics {
 		}
 	}
 
+	if _, ok := g.d.GetOk("compress"); ok { //if true, compress directory name of where the export is going to occur
+		// read all the files
+		var files []fileMeta
+		ferr := filepath.Walk(g.exportDirPath, func(path string, info os.FileInfo, ferr error) error {
+			files = append(files, fileMeta{Path: path, IsDir: info.IsDir()})
+			return nil
+		})
+		if ferr != nil {
+			log.Fatalln(ferr)
+		}
+		// create a zip
+		archive, ferr := os.Create("../archive.zip")
+		if ferr != nil {
+			panic(ferr)
+		}
+		defer archive.Close()
+		zipWriter := zip.NewWriter(archive)
+
+		for _, f := range files {
+			if !f.IsDir {
+				fPath := f.Path
+
+				w, ferr := zipWriter.Create(path.Base(fPath))
+				if ferr != nil {
+					log.Fatalln(ferr)
+				}
+
+				file, ferr := os.Open(f.Path)
+				if ferr != nil {
+					log.Fatalln(ferr)
+				}
+				defer file.Close()
+
+				if _, ferr = io.Copy(w, file); ferr != nil {
+					log.Fatalln(ferr)
+				}
+			}
+		}
+		zipWriter.Close()
```

Comment:
I would move all the above into its own function. ~40 lines is a lot to add to the function `generateOutputFiles`. 

I would also return `diag.Diagnostics` from that new function, and anywhere where you have `log.Fatalln` or `panic` , change it to `return diag.Errorf("...")`. If the new function returns an error, `generateOutputFiles` should then return it. We don't panic/exit anywhere in our code. We try to pass the error back up the tree